engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -448,6 +450,14 @@ public void startApplication() throws YarnException, IOException
       }
       String libJarsCsv = copyFromLocal(fs, appPath, localJarFiles.toArray(new String[]{}));
 
+      if (UserGroupInformation.isSecurityEnabled() && customKeystoreConfig != null && customKeystorePath != null) {
+        String[] customSSLConfArr = {customKeystoreConfig, customKeystorePath};
+        String customSSLFileConf = copyFromLocal(fs, appPath, customSSLConfArr);
+        String configFileName = (new java.io.File(customKeystoreConfig)).getName();
+        dag.getAttributes().put(Context.DAGContext.STRAM_HTTP_CUSTOM_CONFIG, configFileName);
```

Comment:
I agree that we should also address existing misnomers, but more importantly we should reach agreement how new additions are handled. Consistency cannot be an excuse to create more issues. 

In this case, since the attribute that was added really refers to a YARN specific configuration structure, why is that not reflected in the naming? I think that would be the first consideration, since it should not look to the user as if this was something specified by Apex. One would probably have to look into the YARN documentation anyways to know how to use it.

BTW there were more things to address in this PR, not sure if @devtagare is still working on it.